File: Old/Suggested Questions/dateTime.py
from datetime import datetime
str = "January 01, 2007 12:00"
li = str.split(" ")
str = li[0][:3] + " " + li[1][:-1] + " " + li[2] + " " + li[3]

def yearProgress(currentDate):
    # Write your code here
    date = datetime.strptime(currentDate, '%b %d %Y %I:%M')
    #date=datetime.strptime(currentDate, "%b %d, %Y %I:%M")
    month_name = date.month
    #month_number=datetime.strptime(month_name, '%B').month
    elapsedtime= ((month_name)*(date.day)*24*60)
    if((date.year)%400):
        totaltime= 366*24*60
    else:
        totaltime= 365*24*60
    return (elapsedtime/totaltime)*100

print(yearProgress(str))



'''a = datetime(2007, 1, 1, 12, 00)
b = datetime(2007, 11, 30, 23, 45)

print(a.date().month)

print(a)
print(b)

t = b-a;

print(t)
print(t.days*24*60*60+t.seconds)

print(a)
'''

File: Old/Suggested Questions/dateTime.cpp
#include<bits/stdc++.h>
using namespace std;

int calculate(string s)
{
  string month;

  int i=0;
  while(!isspace(s[i]))
  {
    month+=s[i];
    i++;
  }
  i++;

  string date;
  while(s[i]!=',')
  {
    date+=s[i];
    i++;
  }
  i+=2;

  string year;
  while(!isspace(s[i]))
  {
    year+=s[i];
    i++;
  }
  i++;

  string hh;
  while(s[i]!=':')
  {
    hh+=s[i];
    i++;
  }
  i++;

  string min;
  while(s[i]!='\0')
  {
    min+=s[i];
    i++;
  }
  cout<<month<<" "<<date<<" "<<year<<" "<<hh<<" "<<min<<"\n";

  return 0;
}

int main()
{
  string s = "January 01, 2007 12:00";

  int ans = calculate(s);
  //cout<<ans<<"\n";
  return 0;
}
